### PR TITLE
[Vega] Add Filter custom label for opensearchDashboardsAddFilter

### PR DESCRIPTION
**Pull request by aswath86**
### Description
Vega 'opensearchDashboardsAddFilter' allows to add OpenSearch Query DSL and index. The implementation buildQueryFilter takes in third parameter to set a custom label in 'Add Filter'. At the moment, opensearchDashboardsAddFilter can only take query and index parameters. 
 
### Issues Resolved 
This PR adds in the third 'alias' parameter to set the custom label from Vega

`buildQueryFilter = (query: QueryStringFilter['query'], index: string, alias: string)`

Below image is self explanatory. The first filter in the image is without custom label (ugly, in my opinion), and the second filter is with custom label set in Vega.

Vega_OSDAddFilter

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by joshuarrrr** [comment]
@aswath86 Thanks for the contribution! The automation is failing due to a prettier linting issue - it can likely be autofixed. Can you also add an entry to `CHANGELOG.md`?

File: src/plugins/vis_type_vega/public/vega_view/vega_base_view.js [hunk 1]
```diff
@@ -315,10 +315,11 @@ export class VegaBaseView {
   /**
    * @param {object} query Query DSL snippet, as used in the query DSL editor
    * @param {string} [index] as defined in OpenSearch Dashboards, or default if missing
+   * @param {string} OpenSearch Query DSL's Custom label for opensearchDashboardsAddFilter, as used in '+ Add Filter'
```
Comment (ashwin-pc) [inline src/plugins/vis_type_vega/public/vega_view/vega_base_view.js:318]:
```suggestion
   * @param {string} alias OpenSearch Query DSL's Custom label for opensearchDashboardsAddFilter, as used in '+ Add Filter'
```

**Review by ashwin-pc — COMMENTED** [review]
These changes should fix your lint issues.

**Comment by ananzh** [comment]
@aswath86 1.3.8 release is due this week. Do you have some compacities to work on it this week? or should we update the 1.3.8 label?

**Review by ananzh — DISMISSED** [review]

**Comment by ananzh** [comment]
@aswath86  I see some lint issues. Do you have time to take a look?

```
$ yarn run lint:es && yarn run lint:style
$ node scripts/eslint

Error while parsing /__w/OpenSearch-Dashboards/OpenSearch-Dashboards/src/plugins/vis_type_vega/public/vega_view/vega_base_view.js
Line 415, column 23: Line 415: Missing semicolon.

  413 |               'Error setting time filter: both time values must be either relative or absolute dates. {start}, {end}',
  414 |             values: {
> 415 |               start: `start=${JSON.stringify(start)}`,
      |                       ^
  416 |               end: `end=${JSON.stringify(end)}`,
  417 |             },
  418 |           })
`parseForESLint` from parser `/__w/OpenSearch-Dashboards/OpenSearch-Dashboards/node_modules/babel-eslint/lib/index.js` is invalid and will just be ignored

/__w/OpenSearch-Dashboards/OpenSearch-Dashboards/src/plugins/vis_type_vega/public/data_model/vega_parser.test.js
  34:40  error  Parse errors in imported module '../vega_view/vega_base_view': Line 415: Missing semicolon.

  413 |               'Error setting time filter: both time values must be either relative or absolute dates. {start}, {end}',
  414 |             values: {
> 415 |               start: `start=${JSON.stringify(start)}`,
      |                       ^
  416 |               end: `end=${JSON.stringify(end)}`,
  417 |             },
  418 |           }) (415:[23](https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/4008649911/jobs/6883077918#step:9:24))  import/namespace

/__w/OpenSearch-Dashboards/OpenSearch-Dashboards/src/plugins/vis_type_vega/public/vega_view/vega_base_view.js
  415:23  error  Parsing error: Missing semicolon.

  413 |               'Error setting time filter: both time values must be either relative or absolute dates. {start}, {end}',
  414 |             values: {
> 415 |               start: `start=${JSON.stringify(start)}`,
      |                       ^
  416 |               end: `end=${JSON.stringify(end)}`,
  417 |             },
  418 |           })

/__w/OpenSearch-Dashboards/OpenSearch-Dashboards/src/plugins/vis_type_vega/public/vega_view/vega_map_view.js
  33:[30](https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/4008649911/jobs/6883077918#step:9:31)  error  Parse errors in imported module './vega_base_view': Line 415: Missing semicolon.

  413 |               'Error setting time filter: both time values must be either relative or absolute dates. {start}, {end}',
  414 |             values: {
> 415 |               start: `start=${JSON.stringify(start)}`,
      |                       ^
  416 |               end: `end=${JSON.stringify(end)}`,
  417 |             },
  418 |           }) (415:23)  import/namespace

/__w/OpenSearch-Dashboards/OpenSearch-Dashboards/src/plugins/vis_type_vega/public/vega_view/vega_view.js
  [32](https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/4008649911/jobs/6883077918#step:9:33):30  error  Parse errors in imported module './vega_base_view': Line 415: Missing semicolon.

  413 |               'Error setting time filter: both time values must be either relative or absolute dates. {start}, {end}',
  414 |             values: {
> 415 |               start: `start=${JSON.stringify(start)}`,
      |                       ^
  416 |               end: `end=${JSON.stringify(end)}`,
  [41](https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/4008649911/jobs/6883077918#step:9:42)7 |             },
  418 |           }) (415:23)  import/namespace

✖ 4 problems (4 errors, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

**Comment by ananzh** [comment]
I removed v1.3.8 tag since it is not a bug fix.

**Comment by aswath86** [comment]
> @aswath86 Though I approved for the contents, I see some lint issues when checking the tests. Do you have time to take a look?

Hello @ananzh I realised the lint issue is due some markdown syntax that shouldn't be there. Have cleaned them up now. Thank you!

**Comment by AMoo-Miki** [comment]
@aswath86, even though you signed you commits with your `gmail` address, because it is not publicly attached to your account, DCO is failing. I suspect you have "Keep my email addresses private" checked on you [profile's email settings](https://github.com/settings/emails).

**Comment by aswath86** [comment]
@AMoo-Miki Unchecked. Thank you!

**Pull request closed without merging**